libraries/ms-helper/src/generic_can_uart.c

```diff
@@ -28,7 +28,6 @@ static void prv_generic_can_uart_rx_handler(const CanUart *can_uart, uint32_t id
         .dlc = dlc,
       };
       gcu->base.rx_storage[i].rx_handler(&msg, gcu->base.rx_storage[i].context);
-      break;
```

Comment:
This will cause multiple RX handlers to run from a single CAN message

Reply:
I don't think it's a good idea to modify library behavior for testing. It also results in different behavior from the generic CAN network layer implementation.

Reply:
AFAIK nothing relies on only 1 handler running per message?

Reply:
I'm OK with this being inconsistent since Generic CAN Network is highly niche. I have no reliance on the behavior. However it is a significant paradigm shift from how the other CAN Network handlers run which might be confusing to other users. Maybe add comments about this?

Reply:
Basically, the only reason why I made this change is so that updating the appropriate assertion function doesn't rely on using flag variables that signal what assertions to run.

I could set an expected global state `enum` variable or something in the test body, and then use that in the callback to determine which assertions to run? I think that *might* work around needing these changes?

Otherwise, I guess we could implement overriding/removing Generic CAN RX handlers?

Reply:
Shouldn't you be able to set handlers for messages destined to `MOTOR_CONTROLLER_INTERFACE_LEFT_ADDR` and `MOTOR_CONTROLLER_INTERFACE_RIGHT_ADDR`? The messages you're sending shouldn't collide with any registered RX handlers.

Reply:
Not really?

In order to assert on state, you'd need to assert on the data inside the message, since the only messages you'd be sending out are the torque and power messages (which would all have the same message id).

Reply:
But that has nothing to do with this requirement - the reason you need to support multiple RX handlers that match the same mask is because you use a fully accepting mask/filter. If you just set it to messages that come from `MOTOR_CONTROLLER_INTERFACE_LEFT_ADDR` and `MOTOR_CONTROLLER_INTERFACE_RIGHT_ADDR`, you shouldn't need this change.

Reply:
ping